build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/DependenciesGraphTask.java

```diff
@@ -85,55 +85,55 @@ void generateDependenciesGraph() {
-                .append(versionedId)
-                .append("\",\"pkgId\": \"")
-                .append(versionedId)
-                .append("\",\"deps\": []}");
+            packages.add("""
+                {"id": "%s","info": {"name": "%s","version": "%s"}}\
+                """.formatted(versionedId, id, dependency.getVersion()));
+            nodeString.append("""
+                {"nodeId": "%s","pkgId": "%s","deps": []}\
+                """.formatted(versionedId, versionedId));
             nodes.add(nodeString.toString());
-            nodeIds.add("{\"nodeId\": \"" + versionedId + "\"}");
+            nodeIds.add("""
+                {"nodeId": "%s"}\
+                """.formatted(versionedId));
         }
         // We add one package and one node for each dependency, it suffices to check packages.
         if (packages.size() > 0) {
             final String projectName = "elastic/elasticsearch" + getProject().getPath();
-            final StringBuilder output = new StringBuilder();
-            output.append("{\"depGraph\": {\"schemaVersion\": \"1.2.0\",\"pkgManager\": {\"name\": \"gradle\"},\"pkgs\": [")
-                .append("{\"id\": \"")
-                .append(projectName)
-                .append("@0.0.0")
-                .append("\", \"info\": {\"name\": \"")
-                .append(projectName)
-                .append("\", \"version\": \"0.0.0\"}},")
-                .append(String.join(",", packages))
-                .append("],\"graph\": {\"rootNodeId\": \"")
-                .append(projectName)
-                .append("@0.0.0")
-                .append("\",\"nodes\": [")
-                .append("{\"nodeId\": \"")
-                .append(projectName)
-                .append("@0.0.0")
-                .append("\",\"pkgId\": \"")
-                .append(projectName)
-                .append("@0.0.0")
-                .append("\",\"deps\": [")
-                .append(String.join(",", nodeIds))
-                .append("]},")
-                .append(String.join(",", nodes))
-                .append("]}}}");
-            getLogger().debug("Dependency Graph: " + output.toString());
+            final String output = """
+                {
+                  "depGraph": {
+                    "schemaVersion": "1.2.0",
+                    "pkgManager": {"name": "gradle"},
+                    "pkgs": [
+                      {
+                        "id": "%s@0.0.0",
+                        "info": {"name": "%s", "version": "0.0.0"}
+                      },
+                      %s
+                    ],
+                    "graph": {
+                      "rootNodeId": "%s@0.0.0",
+                      "nodes": [
+                        { "nodeId": "%s@0.0.0","pkgId": "%s@0.0.0","deps": [%s] },
+                        %s
+                      ]
+                    }
+                  }
+                }""".formatted(
+                projectName,
```

Comment:
It's not essential, but you could only supply `projectName` once here, and refer to it in the format string with e.g. `%1$s`